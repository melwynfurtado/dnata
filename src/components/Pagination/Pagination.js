import React from 'react'
import PropTypes from 'prop-types'

const Pagination = ({ hotelCount, page, perPage, handleOnPaginate }) => {
  const totalPages = Math.ceil(hotelCount / perPage)
  let pageLinks = []

  for(let i=page; i<=totalPages && i<page+3; i++) {
    pageLinks.push(
      <li key={i} className={`page-item page-item-${i} ${page === i ? 'active': ''}`}>
        <a className="page-link" href="#" page={i} onClick={handleOnPaginate}>{i}</a>
      </li>
    )
  }

  return (
    <nav>
      <ul className="pagination float-right">
        <li key="prev" className={`page-item prev ${page === 1 ? 'disabled': ''}`}>
          <a className="page-link" href="#" page={page-1} aria-label="Previous" onClick={handleOnPaginate}>
            <span aria-hidden="true" page={page-1} onClick={handleOnPaginate}>&laquo;</span>
            <span className="sr-only">Previous</span>
          </a>
        </li>
        { pageLinks }
        <li key="next" className={`page-item next ${page === totalPages ? 'disabled': ''}`}>
          <a className="page-link" href="#" page={page+1} aria-label="Next" onClick={handleOnPaginate}>
            <span aria-hidden="true" page={page+1} onClick={handleOnPaginate}>&raquo;</span>
            <span className="sr-only">Next</span>
          </a>
        </li>
      </ul>
      <div className="font-italic text-muted">{`Page: ${page} / ${totalPages}`}</div>
    </nav>
  )
}

Pagination.propTypes = {
  handleOnPaginate: PropTypes.func,
  hotelCount: PropTypes.number,
  page: PropTypes.number,
  perPage: PropTypes.number,
}

export default Pagination