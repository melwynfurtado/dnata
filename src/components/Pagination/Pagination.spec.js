import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Pagination from './Pagination'

describe('Pagination Component', () => {
  let handleOnPaginateMock
  let Component

  beforeEach(() => {
    handleOnPaginateMock = jest.fn()
    Component = <Pagination 
      hotelCount={35} 
      page={1}
      perPage={20}
      handleOnPaginate={handleOnPaginateMock}
    />
  })

  it('renders correctly', () => {
    const tree = renderer
                  .create(Component)
                  .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls handleOnPaginate onclick of next', () => {
    const wrapper = shallow(Component)
    wrapper.find('.pagination .next a').simulate('click')
    expect(handleOnPaginateMock).toHaveBeenCalled()
  })

  it('does not call handleOnPaginate onclick of prev (disabled)', () => {
    const wrapper = shallow(Component)
    wrapper.find('.pagination .prev').simulate('click')
    expect(handleOnPaginateMock).not.toHaveBeenCalled()
  })  

  it('calls handleOnPaginate onclick of second page', () => {
    const wrapper = shallow(Component)
    wrapper.find('.pagination .page-item-2 a').simulate('click')
    expect(handleOnPaginateMock).toHaveBeenCalled()
  }) 
})