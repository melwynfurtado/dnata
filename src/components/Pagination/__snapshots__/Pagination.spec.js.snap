// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component renders correctly 1`] = `
<nav>
  <ul
    className="pagination float-right"
  >
    <li
      className="page-item prev disabled"
    >
      <a
        aria-label="Previous"
        className="page-link"
        href="#"
        onClick={[MockFunction]}
        page={0}
      >
        <span
          aria-hidden="true"
          onClick={[MockFunction]}
          page={0}
        >
          «
        </span>
        <span
          className="sr-only"
        >
          Previous
        </span>
      </a>
    </li>
    <li
      className="page-item page-item-1 active"
    >
      <a
        className="page-link"
        href="#"
        onClick={[MockFunction]}
        page={1}
      >
        1
      </a>
    </li>
    <li
      className="page-item page-item-2 "
    >
      <a
        className="page-link"
        href="#"
        onClick={[MockFunction]}
        page={2}
      >
        2
      </a>
    </li>
    <li
      className="page-item next "
    >
      <a
        aria-label="Next"
        className="page-link"
        href="#"
        onClick={[MockFunction]}
        page={2}
      >
        <span
          aria-hidden="true"
          onClick={[MockFunction]}
          page={2}
        >
          »
        </span>
        <span
          className="sr-only"
        >
          Next
        </span>
      </a>
    </li>
  </ul>
  <div
    className="font-italic text-muted"
  >
    Page: 1 / 2
  </div>
</nav>
`;
